### PR TITLE
Offline Micro Gateway with API key

### PR DESCRIPTION
**Pull request by sabeena1217**
- Enables the offline micro api gateway, with the API key
(can run the gateway without any connection with the APIM core)

### CODE REVIEW

File: gateway/conf/deployment.yaml [hunk 1]
```diff
@@ -36,16 +36,17 @@ ballerina.config:
...
+#            # Package name of the request interceptor.
+#            packageName: org.wso2.carbon.apimgt.gateway.throttle
+#            # Ballerina Archive name.
+#            archiveName: throttle.bmz
+#          -
+#            # Package name of the request interceptor.
+#            packageName: org.wso2.carbon.apimgt.gateway.handler
+#            # Ballerina Archive name.
+#            archiveName: analytics.bmz
+
```
Comment (pubudu538) [inline gateway/conf/deployment.yaml]: It is better to remove those commented configurations. We can update the docs on how to configure deployment.yaml when starting as the offline gateway.

File: gateway/microgateway/README.txt [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+               "name":"",
+               "sandbox":"",
+               "production":""
+            }
+         ]
+      }
+   ]
+}
+3. Include all the API.bal files in the microgateway folder.
+4. (optional) If you wish to start the gateway with analytics and throttling, uncomment the analytics and throttling in the gateway deployment.yaml file and include the 'microConf.json' where the data is in the following format.
```
Comment (pubudu538) [inline gateway/microgateway/README.txt]: We can say what to put there in the deployment.yaml.

File: gateway/conf/deployment.yaml (hunk 1 above)
Comment (sabeena1217) [reply gateway/conf/deployment.yaml]: noted

File: gateway/microgateway/README.txt (hunk 2 above)
Comment (sabeena1217) [reply gateway/microgateway/README.txt]: noted

File: integration-tests/src/main/java/org/wso2/carbon/apimgt/rest/integration/tests/microgateway/ApiClient.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
+package org.wso2.carbon.apimgt.rest.integration.tests.microgateway;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.*;
+//import javax.net.ssl.SSLSocketFactory;
```
Comment (mushthaq33) [inline integration-tests/src/main/java/org/wso2/carbon/apimgt/rest/integration/tests/microgateway/ApiClient.java]: Let's remove all the commented out code.

File: integration-tests/src/main/java/org/wso2/carbon/apimgt/rest/integration/tests/microgateway/ApiClient.java [hunk 4]
```diff
@@ -0,0 +1,165 @@
+package org.wso2.carbon.apimgt.rest.integration.tests.microgateway;
```
Comment (mushthaq33) [inline integration-tests/src/main/java/org/wso2/carbon/apimgt/rest/integration/tests/microgateway/ApiClient.java]: License header missing

File: integration-tests/src/test/java/org/wso2/carbon/apimgt/rest/integration/tests/microgateway/ApiClientTest.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            Assert.assertEquals(response, "{ \"abc\": \"This is a test\" }", "API name mismatch");
+        } catch (NoSuchAlgorithmException e) {
+            //log.error("Failed to initialize SSL handling.", e);
+        } catch (KeyManagementException e) {
+            //log.error("Failed to initialize SSL handling.", e);
+        }
+        //Assert.assertEquals(response, "\"abc\": \"This is a test\"", "API name mismatch");
+    }
+
+//    @Test
```
Comment (mushthaq33) [inline integration-tests/src/test/java/org/wso2/carbon/apimgt/rest/integration/tests/microgateway/ApiClientTest.java]: Let's remove the commented out code

**Pull request closed without merging**